src/appleseed/renderer/modeling/phasefunction/henyeyphasefunction.cpp

```diff
@@ -168,13 +168,13 @@ class HenyeyPhaseFunction
         }
         else
         {
-            const float t = (1.0f - sqr_g) / (1.0f + g*s);
+            const float t = (1.0f - sqr_g) / (1.0f + g * s);
             cosine = 0.5f / g * (1.0f + sqr_g - t * t);
         }
-        const float sine = std::sqrt(1.0f - cosine * cosine);
 
-        const Vector2f tangent =
-            sample_circle_uniform(sampling_context.next2<float>());
+        const float sine = std::sqrt(clamp(1.0f - cosine * cosine, 0.0f, 1.0f));
```

Comment:
You can also use `foundation::saturate()` which is equivalent to `foundation::clamp(x, 0.0f, 1.0f)`. Both are fine.